src/software/SfM/main_rigTransform.cpp

```diff
@@ -0,0 +1,127 @@
+#include <openMVG/rig/Rig.hpp>
+#include <openMVG/logger.hpp>
+
+#include <openMVG/sfm/sfm_data.hpp>
+#include <openMVG/sfm/sfm_data_io.hpp>
+
+#if HAVE_ALEMBIC
+#include <openMVG/sfm/AlembicExporter.hpp>
+#endif // HAVE_ALEMBIC
+
+#include <boost/filesystem.hpp>
+#include <boost/progress.hpp>
+#include <boost/program_options.hpp> 
+#include <iostream>
+#include <string>
+#include <chrono>
+#include <memory>
+
+namespace bfs = boost::filesystem;
+namespace po = boost::program_options;
+
+using namespace openMVG;
+using namespace openMVG::sfm;
+
+static std::vector<double> ReadIntrinsicsFile(const std::string& fname)
+{
+  cout << "reading intrinsics: " << fname << endl;
+
+  std::vector<double> v(8);
+  std::ifstream ifs(fname);
+  if (!(ifs >> v[0] >> v[1] >> v[2] >> v[3] >> v[4] >> v[5] >> v[6] >> v[7]))
+    throw std::runtime_error("failed to read intrinsics file");
+  return v;
+}
+
+int main(int argc, char** argv)
+{
+  std::string exportFile = "trackedcameras-rig.abc"; //!< the export file
+  std::string importFile = "";
+  std::string rigFile = "";
+  std::string calibFile = "";
+  std::vector<openMVG::geometry::Pose3> extrinsics;  // the rig subposes
+  
+  po::options_description desc("This program is used to transform cameras localized with a rig file.");
+  desc.add_options()
+        ("help,h", "Print this message")
+        ("input,i", po::value<std::string>(&importFile)->required(),
```

Comment:
Since we use `Load()` it can be any of the supported files (json, abc etc), I think. Not sure for animated cameras though.